pkg/confgen/flowlogs2metrics_config.go

```diff
@@ -22,163 +22,70 @@ import (
-				Follows: "ingest_collector",
-			},
-			{Name: "transform_network",
-				Follows: "transform_generic",
-			},
-			{Name: "extract_aggregate",
-				Follows: "transform_network",
-			},
-			{Name: "encode_prom",
-				Follows: "extract_aggregate",
-			},
-			{Name: "write_loki",
-				Follows: "transform_network",
-			},
-		},
-		Parameters: []config.StageParam{
-			{Name: "ingest_collector",
-				Ingest: &config.Ingest{
-					Type: "collector",
-					Collector: &api.IngestCollector{
-						Port:       cg.config.Ingest.Collector.Port,
-						PortLegacy: cg.config.Ingest.Collector.PortLegacy,
-						HostName:   cg.config.Ingest.Collector.HostName,
-					},
-				},
-			},
-			{Name: "transform_generic",
-				Transform: &config.Transform{
-					Type: "generic",
-					Generic: &api.TransformGeneric{
-						Policy: "replace_keys",
-						Rules:  cg.config.Transform.Generic.Rules,
-					},
-				},
-			},
-			{Name: "transform_network",
-				Transform: &config.Transform{
-					Type: "network",
-					Network: &api.TransformNetwork{
-						Rules: cg.transformRules,
-					},
-				},
-			},
-			{Name: "extract_aggregate",
-				Extract: &config.Extract{
-					Type:       "aggregates",
-					Aggregates: cg.aggregateDefinitions,
-				},
-			},
-			{Name: "encode_prom",
-				Encode: &config.Encode{
-					Type: "prom",
-					Prom: &api.PromEncode{
-						Port:    cg.config.Encode.Prom.Port,
-						Prefix:  cg.config.Encode.Prom.Prefix,
-						Metrics: cg.promMetrics,
-					},
-				},
-			},
-			{Name: "write_loki",
-				Write: &config.Write{
-					Type: cg.config.Write.Type,
-					Loki: &cg.config.Write.Loki,
-				},
-			},
-		},
+func (cg *ConfGen) GenerateFlowlogs2PipelineConfig() map[string]interface{} {
+	pipeline, _ := config.NewPipeline("ingest_collector", &cg.config.Ingest)
+	next := pipeline
+	if cg.config.Transform.Generic != nil {
+		gen := *cg.config.Transform.Generic
+		if len(gen.Policy) == 0 {
+			gen.Policy = "replace_keys"
+		}
+		next = next.TransformGeneric("transform_generic", gen)
+	}
+	if len(cg.transformRules) > 0 {
+		next = next.TransformNetwork("transform_network", api.TransformNetwork{
+			Rules: cg.transformRules,
+		})
+	}
+	if len(cg.aggregateDefinitions) > 0 {
+		agg := next.Aggregate("extract_aggregate", cg.aggregateDefinitions)
+		agg.EncodePrometheus("encode_prom", api.PromEncode{
+			Port:    cg.config.Encode.Prom.Port,
+			Prefix:  cg.config.Encode.Prom.Prefix,
+			Metrics: cg.promMetrics,
+		})
+	}
+	if cg.config.Write.Loki != nil {
+		next.WriteLoki("write_loki", *cg.config.Write.Loki)
+	}
+	return map[string]interface{}{
+		"log-level":  "error",
+		"pipeline":   pipeline.GetStages(),
+		"parameters": pipeline.GetStageParams(),
 	}
-	return configStruct
 }
 
-func (cg *ConfGen) GenerateTruncatedConfig(stages []string) config.ConfigFileStruct {
-	parameters := make([]config.StageParam, len(stages))
-	for i, stage := range stages {
+func (cg *ConfGen) GenerateTruncatedConfig() []config.StageParam {
```

Comment:
This signature change requires changes in NOO.

Reply:
Yes I have this PR on NOO https://github.com/netobserv/network-observability-operator/pull/142